### PR TITLE
[incubator][VC] support tenant aware dns server

### PR DESCRIPTION
**Pull request by zhuangqh**
add tenant dns setup script 

support tenant aware dns server
 1. get tenant name server from service kube-system/kube-dns
 2. mutate pPod's dns config to bypass kubelet's behavior

related to #250

### CODE REVIEW

File: incubator/virtualcluster/pkg/syncer/conversion/mutate.go [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+	}
+	return ret
+}
+
+type ServiceMutateInterface interface {
+	Mutate()
+}
+
+type serviceMutator struct {
+	vc       *v1alpha1.VirtualclusterSpec
```
Comment (Fei-Guo) [inline incubator/virtualcluster/pkg/syncer/conversion/mutate.go]: Does service mutator need VC at all?

Comment (zhuangqh) [reply incubator/virtualcluster/pkg/syncer/conversion/mutate.go]: nope. But maybe it does later. This is a unify framework for mutator

File: incubator/virtualcluster/pkg/syncer/cluster/cluster.go [hunk 2]
```diff
@@ -49,6 +50,9 @@ type Cluster struct {
 	// Config is the rest.config used to talk to the apiserver.  Required.
 	Config *rest.Config
 
+	// spec is the vc definition. Required.
+	spec *v1alpha1.VirtualclusterSpec
```
[the remark this answers is not in the document]
Comment (zhuangqh) [reply incubator/virtualcluster/pkg/syncer/cluster/cluster.go]: we may use other fields later

**Pull request merged**